### PR TITLE
Add support for Managed AD Peering

### PR DESCRIPTION
**Pull request by Jberlinsky**
Adds support for Managed Active Directory Domain Peering. Fixes https://github.com/hashicorp/terraform-provider-google/issues/12119

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests. **Note -- There appear to be lint errors, but they are in code paths that I did not modify**
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_active_directory_peering: Added support for [Domain Peering](https://cloud.google.com/managed-microsoft-ad/docs/domain-peering)(beta)
```

### CODE REVIEW

**Comment by Jberlinsky** [comment]
@megan07 In reviewing some of the test output, it appears that the testing project/org is exceeding its Managed Active Directory quota. Any chance this can be raised?

**Comment by Jberlinsky** [comment]
@megan07 Marking this ready for review -- the test runs:

* For AD-related resources, look like quota issues that are outside my control. Can you assist with that?
* For non-AD related resources, I'm not sure how I could have meaningfully impacted those.

**Review by megan07 — APPROVED** [review]
A couple minor comments, but otherwise looks good! It passed locally for me, so should be good to go after a quick update! Thanks!

File: mmv1/products/activedirectory/api.yaml
Comment (Jberlinsky) [reply mmv1/products/activedirectory/api.yaml]: Fixed!

File: mmv1/products/activedirectory/api.yaml
Comment (Jberlinsky) [reply mmv1/products/activedirectory/api.yaml]: Removed

**Comment by Jberlinsky** [comment]
@megan07 This is ready for re-review :)

**Comment by megan07** [comment]
I re-ran it locally to double-check and it passed again:

>=== RUN   TestAccActiveDirectoryPeering_activeDirectoryPeeringBasicExample
=== PAUSE TestAccActiveDirectoryPeering_activeDirectoryPeeringBasicExample
=== CONT  TestAccActiveDirectoryPeering_activeDirectoryPeeringBasicExample
--- PASS: TestAccActiveDirectoryPeering_activeDirectoryPeeringBasicExample (2762.63s)
PASS

Thanks!

**Comment by Jberlinsky** [comment]
@megan07 I'm not seeing this in the [changelog](https://github.com/hashicorp/terraform-provider-google-beta/blob/main/CHANGELOG.md) for the `google-beta` provider, nor in the Hashicorp-hosted docs, despite releases in the changelog dated after this was merged. Everything alright with this change?

**Comment by megan07** [comment]
Hi @Jberlinsky ! We usually cut the release a few days prior to the actual release date, so although it was merged before the release it is slated for the next release. Thanks!

**Pull request merged**
